it/uk/gov/hmrc/breathingspaceifproxy/connector/IndividualDetailsConnectorISpec.scala

```diff
@@ -0,0 +1,70 @@
+
+  val connector = inject[IndividualDetailsConnector]
+
+  "getMinimalPopulation" should {
+    "return a MinimalPopulation instance when it receives a 200(OK) response" in {
+      val nino = genNino
+      val url = urlWithoutQuery(IndividualDetailsConnector.path(nino, IndividualDetailsConnector.minimalPopulation))
+      val responsePayload = Json.toJson(individualDetailsMinimalResponse(nino.value)).toString
+      stubCall(HttpMethod.Get, url, Status.OK, responsePayload, minimalPopulation)
+
+      val response = await(connector.getMinimalPopulation(nino))
+
+      verifyHeaders(HttpMethod.Get, url, minimalPopulation.head)
+      assert(response.fold(_ => false, _ => true))
+    }
+
+    "return RESOURCE_NOT_FOUND when the provided resource is unknown" in {
+      val unknownNino = genNino
+      val url = urlWithoutQuery(IndividualDetailsConnector.path(unknownNino, IndividualDetailsConnector.minimalPopulation))
+      stubCall(HttpMethod.Get, url, Status.NOT_FOUND, errorResponsePayloadFromIF, minimalPopulation)
+
+      val response = await(connector.getMinimalPopulation(unknownNino))
+
+      verifyHeaders(HttpMethod.Get, url, minimalPopulation.head)
+      response.fold(_.head.baseError shouldBe RESOURCE_NOT_FOUND, _ => notAnErrorInstance)
+    }
+
+    "return CONFLICTING_REQUEST in case of duplicated requests" in {
+      val nino = genNino
+      val url = urlWithoutQuery(IndividualDetailsConnector.path(nino, IndividualDetailsConnector.minimalPopulation))
+      stubCall(HttpMethod.Get, url, Status.CONFLICT, errorResponsePayloadFromIF, minimalPopulation)
+
+      val response = await(connector.getMinimalPopulation(nino))
+
+      verifyHeaders(HttpMethod.Get, url, minimalPopulation.head)
+      response.fold(_.head.baseError shouldBe CONFLICTING_REQUEST, _ => notAnErrorInstance)
+    }
+
+    "return SERVER_ERROR for any 4xx error, 404 and 409 excluded" in {
+      val nino = genNino
+      val url = urlWithoutQuery(IndividualDetailsConnector.path(nino, IndividualDetailsConnector.minimalPopulation))
+      stubCall(HttpMethod.Get, url, Status.BAD_REQUEST, errorResponsePayloadFromIF, minimalPopulation)
+
+      val response = await(connector.getMinimalPopulation(nino))
+
+      verifyHeaders(HttpMethod.Get, url, minimalPopulation.head)
+      response.fold(_.head.baseError shouldBe SERVER_ERROR, _ => notAnErrorInstance)
+    }
+
+    "return SERVER_ERROR for any 5xx error" in {
+      val nino = genNino
+      val url = urlWithoutQuery(IndividualDetailsConnector.path(nino, IndividualDetailsConnector.minimalPopulation))
+      stubCall(HttpMethod.Get, url, Status.BAD_GATEWAY, errorResponsePayloadFromIF, minimalPopulation)
+
+      val response = await(connector.getMinimalPopulation(nino))
+
+      verifyHeaders(HttpMethod.Get, url, minimalPopulation.head)
+      response.fold(_.head.baseError shouldBe SERVER_ERROR, _ => notAnErrorInstance)
+    }
+  }
+}
```

Comment:
Add a test to check handling a 422 error perhaps

Reply:
We do not have any specific case in `connector.ConnectorHelper` for 422. It's just included, like any other 4xx error (excluding 404 & 409 which are instead specific), in the general Upstream4xxResponse case. There's no point in asking to WireMock to send 422 and then test that WireMock has actually sent a 422.